app/assets/javascripts/discourse/lib/utilities.js.es6

```diff
@@ -410,5 +410,46 @@ export function rescueThemeError(name, error, api) {
   document.body.prepend(alertDiv);
 }
 
+const CODE_BLOCKS_RULES = [
+  { rule: "^(    |\\t).*", end: "" },
+  { rule: "`.+?`", end: "`" },
+  { rule: "^```[^]*?^```", end: "\n```" },
+  { rule: "\\[code\\][^]*?\\[\\/code\\]", end: "\n[/code]" }
+];
+
+export function inCodeBlock(text, pos) {
+  const regexp = new RegExp(
+    CODE_BLOCKS_RULES.map(rules => rules.rule).join("|"),
+    "gm"
+  );
```

Comment:
minor... but why map this every time you run the function, just do it once upfront ?

Reply:
You are right. There is also no point in keeping the array of rules anymore. :)